### ISSUE TITLE
als.cu(205): error: more than one instance of overloaded function "isnan" matches the argument list

### ISSUE THREAD

**Issue by dkchae** [wei-tan/CuMF#15]
Hi.
I am using Linux
I uncommented in als.cu the line #define SURPASS_NAN
and build by
make clean build

But encountered error messages as follows:

rm -f host_utilities.o device_utilities.o als.o main main.o cg.o
/usr/local/cuda/bin/nvcc -ccbin g++   -m64  -std=c++11   -Xcompiler -DADD_   -gencode arch=compute_35,code=sm_35 -gencode arch=compute_35,code=compute_35 -lineinfo -o host_utilities.o -c host_utilities.cpp
/usr/local/cuda/bin/nvcc -ccbin g++   -m64  -std=c++11   -Xcompiler -DADD_   -gencode arch=compute_35,code=sm_35 -gencode arch=compute_35,code=compute_35 -lineinfo -o device_utilities.o -c device_utilities.cu
/usr/local/cuda/bin/nvcc -ccbin g++   -m64  -std=c++11   -Xcompiler -DADD_   -gencode arch=compute_35,code=sm_35 -gencode arch=compute_35,code=compute_35 -lineinfo -o cg.o -c cg.cu
/usr/local/cuda/bin/nvcc -ccbin g++   -m64  -std=c++11   -Xcompiler -DADD_   -gencode arch=compute_35,code=sm_35 -gencode arch=compute_35,code=compute_35 -lineinfo -o als.o -c als.cu
als.cu(205): error: more than one instance of overloaded function "isnan" matches the argument list:
            function "isnan(float)"
            function "std::isnan(float)"
            argument types are: (float)

als.cu(205): error: more than one instance of overloaded function "isnan" matches the argument list:
            function "isnan(float)"
            function "std::isnan(float)"
            argument types are: (float)

2 errors detected in the compilation of "/tmp/tmpxft_00006c73_00000000-7_als.cpp1.ii".

Could you please help me?
Thanks

**Comment by wei-tan** [comment]
Hello, 
First, #define SURPASS_NAN is used when you will probably have NAN in calculating RMSE. E.g., you have test set that does not show up in your training set. Are you sure you need this?

I tried but failed to reproduce your issue. I googled and found this:
http://stackoverflow.com/questions/33770374/why-is-isnan-ambiguous-and-how-to-avoid-it

Please let me know if it still does not work for you.

**Comment by Makazone** [comment]
Had same issue, replacing check in line 205 with `if(a!=a||b!=b)` fixed it.

**Comment by wei-tan** [comment]
Thank you @Makazone. Fixed and pushed the change.
